src/app/planner/components/planner.component.ts

```diff
@@ -0,0 +1,125 @@
+import * as fromRoomActions from '../../rooms/actions/room.action';
+import { ResourceList } from '../../core/callendar/model/resource-list';
+import { IcalendarTransformer } from '../../core/callendar/transformers/icalendar-transformer';
+import { DataWithSimpleResourceTransformer } from '../../core/callendar/transformers/data-with-simple-resource-transformer';
+import { SessionViewModel } from '../../sessions/models/session.viewmodel';
+import { Judge } from '../../judges/models/judge.model';
+import * as judgeActions from '../../judges/actions/judge.action';
+import { Subscription } from 'rxjs/Subscription';
+import { CallendarComponent } from '../../core/callendar/components/callendar.component';
+
+@Component({
+    selector: 'app-planner',
+    templateUrl: './planner.component.html',
+    styleUrls: ['./planner.component.scss']
+})
+export class PlannerComponent implements OnInit {
+
+    header: any;
+    views: any;
+    columns: any[];
+    resources: any[];
+    dataTransformer: IcalendarTransformer<SessionViewModel>;
+    loadDataAction: (SessionQueryForDates) => void;
+    sessions$: Observable<any[]>;
+    defaultView: string;
+    @ViewChild(CallendarComponent) private _slCalendar: CallendarComponent;
+
+    private roomsSubscription: Subscription = Subscription.EMPTY;
+    private judgesSubscription: Subscription = Subscription.EMPTY;
+
+    constructor(private store: Store<State>, private route: ActivatedRoute,
+                public dialog: MatDialog) {
+        this.defaultView = 'timelineWeek';
+        this.header = {
+            left: 'prev,next today',
+            center: 'title',
+            right: 'timelineDay,timelineWeek,timelineMonth'
+        };
+        this.views = {
+            timelineDay: {
+                slotDuration: '00:10'
+            },
+            timelineWeek: {
+                slotDuration: '00:30'
+            }
+        };
+    }
+
+    ngOnInit() {
+        this.sessions$ = this.store.select(fromReducer.getFullSessions);
+        this.route.data.subscribe((data) => {
+            this.loadDataAction = this.loadDataForAllJudges;
+        });
+        this.configureRoomView();
+    }
+
+    public configureRoomView() {
+        this.judgesSubscription.unsubscribe();
+        this.dataTransformer = new DataWithSimpleResourceTransformer('room');
+        this.columns = [
+            {
+                labelText: 'Room',
+                field: 'title'
+            }
+        ];
+        this.roomsSubscription = this.store.pipe(select(fromSessions.getRooms)).subscribe(elements => {
+            let newResourceList = new ResourceList();
+            Object.values(elements).forEach((room: Room) => {
+                newResourceList.add('room-' + room.id, room.name);
+            });
+            this.resources = newResourceList.get();
+        });
+        this.store.dispatch(new fromRoomActions.Get());
+        this._slCalendar.refreshViewData();
```

Comment:
It looks like we refreshViewData immediately after executing a backend request for judges. Shouldn't we rather subscribe to action 'fromRoomActions.GetComplete' and refresh the view after that?